Add netkit telnet

terasawa-lin opened the pull request:
netkit-telnet: Add netkit-telnet package
    
debian's source is applied some patches, so removed the following patches:
0001-telnet-telnetd-Fix-deadlock-on-cleanup.patch
0001-telnet-telnetd-Fix-print-format-strings.patch
To-aviod-buffer-overflow-in-telnet.patch

Modifications to the recipe
remove:
SRC_URI[md5sum]
SRC_URI[sha256sum]
UPSTREAM_CHECK_URI
UPSTREAM_CHECK_REGEX

change:
SRC_URI

# Purpose of pull request

This pr adds netkit-telnet package based on meta-openembedded/meta-networking/recipes-netkit/netkit-telnet/netkit-telnet_0.17.bb.

# Test
## How to install
Add the following to conf/local.conf.
```
IMAGE_INSTALL_append = " xinetd"
IMAGE_INSTALL_append = " netkit-telnet"
```

## How to test

Run following command from console

```
# vi /etc/xinetd.d/telnet
disable = yes
↓
disable = no

# /etc/init.d/xinetd restart
```
Login to localhost using telnet.
If you are prompted to log in to the tested environment, you are successful.

## Test result

```
# telnet 127.0.0.1
Trying 127.0.0.1...
Connected to 127.0.0.1.
Escape character is '^]'.
EMLinux 2.2 ek874

ek874 login:
```

masami256 commented on recipes-debian/xinetd/xinetd_debian.bb:
```diff
@@ -0,0 +1,87 @@
+# base recipe: poky/meta/recipes-extended/xinetd/xinetd_2.3.15.bb
```
This PR/commit should not include xinted's recipes.

terasawa-lin commented:
Removed xinetd.

masami256 commented on recipes-debian/netkit-telnet/netkit-telnet_debian.bb:
```diff
@@ -0,0 +1,67 @@
+LICENSE = "BSD"
+LIC_FILES_CHKSUM = "file://telnet/telnet.cc;beginline=2;endline=3;md5=780868e7b566313e70cb701560ca95ef"
+
+DEBIAN_PATCH_TYPE="quilt"
```
You don't have to set DEBIAN_PATCH_TYPE because quilt is default patch type.

masami256 commented on recipes-debian/netkit-telnet/netkit-telnet_debian.bb:
```diff
@@ -0,0 +1,67 @@
+# base recipe: meta-openembedded/meta-networking/recipes-netkit/netkit-telnet/netkit-telnet_0.17.bb
+# base branch: zeus
```
warrior branch contains netkit-telnet package, is there any reasons to use zeus branch?

terasawa-lin committed:
netkit-telnet: Add netkit-telnet package

debian's source is applied some patches, so removed the following patches:
0001-telnet-telnetd-Fix-deadlock-on-cleanup.patch
0001-telnet-telnetd-Fix-print-format-strings.patch
To-aviod-buffer-overflow-in-telnet.patch

Modifications to the recipe
remove:
SRC_URI[md5sum]
SRC_URI[sha256sum]
UPSTREAM_CHECK_URI
UPSTREAM_CHECK_REGEX

change:
SRC_URI

terasawa-lin commented:
Removed DEBIAN_PATCH_TYPE.
Changed from zeus to use warrior version.

masami256 commented:
@terasawa-lin  thanks!
